### ISSUE TITLE
Cachewriter gone?

### ISSUE THREAD

**Issue by asad-awadia** [ben-manes/caffeine#656]
Hey @ben-manes ,

Where did the CacheWriter method go? On version 3.0.5 there is no `writer` method on `Caffeine.newBuilder()`

like here is an example of how I am trying to use it https://github.com/ben-manes/caffeine/issues/444#issuecomment-671094711  -- but no writer method?

**Comment by ben-manes** [comment]
Hi @asad-awadia,

Unfortunately `CacheWriter` was too broad of an interface that could cause some confusion because it did not fit neatly with the Collections Framework or `AsyncCache`. It was a api design mistake that was much better served with a more narrower feature.

For intercepting explicit writes, the same behavior can be achieved by using Map computations. That is more obvious as not a hidden callback, and by now everyone is comfortable with lambdas and those newer methods.

The unique feature was to synchronously intercept automatic evictions. This was like `RemovalListener` except instead of being called asynchronously after the entry was removed, it was called within the underlying Map's `computeIfPresent` call. That can now be achieved using `Caffeine.evictionListener`.

In summary, we did not remove functionality but did remove a rarely used API that was not always obvious. The replacement advise should be equivalent, clearer, and a little nicer too (as lambda friendly).

Does that help?

**Comment by asad-awadia** [comment]
Ok got it - thank you!
